koinos/protocol/protocol.proto

```diff
@@ -29,6 +29,9 @@ message upload_contract_operation {
    bytes contract_id = 1 [(btype) = CONTRACT_ID];
    bytes bytecode = 2;
    string abi = 3;
+   bool has_authorize_call_contract = 4;
```

Comment:
`authorizes_call_contract` et al. sounds nicer to me -- up to you.

Reply:
The field name will be readable in C++.

Pseudocode looks something like:

``` c++
if ( contract.has_authorize_call_contract() )
{
   ...
}
```

vs

``` c++
if ( contract.authorize_call_contract() )
{
   ...
}
```

which looks better? `has_*` does imo.

Reply:
You forgot the `s`

if contract authorizes call contract is the english.

Just noting that we never use "uses" "has" or any other similar terms in our booleans.